### ISSUE TITLE
Integration Tracking (WC Subscriptions): Export components to a global checkout

### ISSUE THREAD

**Issue by senadir** [woocommerce/woocommerce-gutenberg-products-block#3613]
In order to get #3246 out, we need to a way to export components globally so that they're consumable by other plugins.
Getting an npm package is not a priority right now so getting them to the global `wc` variable should be enough.
The new package should be named checkout and is accessed globally from `wc.checkout`.
When implemented, a new rule should be added to [@woocommerce/dependency-extraction-webpack-plugin](https://github.com/woocommerce/woocommerce-admin/tree/main/packages/dependency-extraction-webpack-plugin) to alias `@woocommerce/checkout` to `wc.checkout`.
I anticipate that wc.checkout would conflict with our block slug, so we should probably name the slug something else when enqueueing the file.
This branch `try/checkout-package` contains an exploration I did some time ago and you can probably build on it or start fresh, up to you.

The new package would contain slotFills and probably some components, so it should be able to build SCSS as well.
We would gradually move components from `assets/js/base/components/cart-checkout` to this package when it makes sense.

**Comment by Aljullu** [comment]
>  I anticipate that wc.checkout would conflict with our block slug, so we should probably name the slug something else when enqueueing the file.

@senadir Is the plan to export something else in addition to components from that package? If not, could we name it `@woocommerce/checkout-components` and `wc.checkoutComponents` to avoid the conflict and make the name more descriptive?

> The new package would contain slotFills and probably some components, so it should be able to build SCSS as well.

Currently, the styles needed for the Checkout components are built into two files `style.css` (which is shared across all blocks) and `vendors-style.css` (which includes some 3rd party styles, ie: `@wordpress/components`). Could we keep it as-is or the plan is to generate a third file?

For now, I would keep it as-is (two files), because the most common use case will be to render those components inside the Checkout block. The benefit is that fewer files will be needed to render the page and, in most cases, those two CSS files will be already in the user browser cache,

If in the future we see Checkout components being used outside of the Checkout block, we can re-evaluate this decision. Sounds good?

**Comment by senadir** [comment]
We discussed naming here pca54o-yB-p2
While initially we're going to share some components, it's also going to house hooks, slotfills and data helpers related to Checkout, so we don't want to rename it later once we publish them.

> Currently, the styles needed for the Checkout components are built into two files style.css (which is shared across all blocks) and vendors-style.css (which includes some 3rd party styles, ie: @wordpress/components). Could we keep it as-is or the plan is to generate a third file?

If we can keep it as is, then I don't have much problem, given that the styles are going to be imported inside those components inside this package.
I'm not sure how technical or complicated this setup is going to get, so having an extra file or everything in `style.css` would depend on how complex the setup is.

**Comment by Aljullu** [comment]
> We discussed naming here pca54o-yB-p2

Aaah thanks. I would like to revisit that discussion again. My issue with naming the package `@woocommerce/checkout` is that its handle conflicts with `wc-checkout` (the Core script used by the Checkout shortcode).

I see the solution you went for in `try/checkout-package` was to use `wc-checkout-packages` as the handle, is that how we are planning to name all packages (`wc-xyz-packages`)? I wonder if it was in purpose not use the singular: `wc-xyz-package` or `wc-package-xyz`. In any case, that sounds like something that should be documented as part of #2660.

Another thing we could try is to name the package `@woocommerce/blocks-checkout` and the handle `wc-blocks-checkout`. This avoids any conflicts with other scripts. If I understand it correctly, everything we expose from that package is intended to be used in extensions that change the C&C blocks or create new blocks, right? In that case, having the prefix `blocks-` helps understand where the package is intended to be used. It would also be in line with `@woocommerce/blocks-registry`.

tl;dr:

* How does it sound naming the package `@woocommerce/blocks-checkout` and the handle `wc-blocks-checkout`?
* Otherwise, what about keeping the package name (`@woocommerce/checkout`) but change the handle from `wc-checkout-packages` to `wc-checkout-package` or `wc-package-checkout`?
  * In this case and if that's a guideline we will follow for all packages, should we add it to #2660?

**Comment by senadir** [comment]
The handle name could be independent from the package alias.
In the future, if we end up releasing several packages then it would make sense to have `@woocommerce/blocks-checkout` `@woocommerce/hosted-checkout` etc so I'm okay with naming it blocks-checkout.
